src/plugins/intel_cpu/src/output_mem_mgr.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "output_mem_mgr.h"
+#include "utils/debug_capabilities.h"
+
+using namespace ov::intel_cpu;
+
+void OutputMemoryMngr::setTensor(std::shared_ptr<Tensor> tensor) {
+    auto _previous = m_pMngr;
+    if (tensor) {
+        auto memptr = tensor->get_memory();
+        m_pMngr = memptr->getMemoryMngr();
+    } else {
+        m_pMngr = m_pOrigMngr;
+    }
+    // WA: resize stage might work because there is no shape change,
+    // but the underlying actual memory manager changes.
+    bool validated = (_previous != m_pMngr);
+    if (validated && m_pMngr->getSize() < m_Size) {
+        m_pMngr->resize(m_Size);
+    }
```

Comment:
@maxnick Here is the WA for https://github.com/maxnick/openvino/pull/62/#discussion_r1244479580 
Sounds not a good solution as well, as it assumes something happening in resizing stage and potential extra resize in some situations.

Reply:
done